hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/Archiver.java

```diff
@@ -115,6 +119,33 @@ public static long includeFile(File file, String entryName,
     return bytes;
   }
 
+  /**
+   * Creates a hardlink for the given file in a temporary directory, adds it
+   * as an entry in the archive, and includes its contents in the archive output.
+   * The temporary hardlink is deleted after processing.
+   */
+  public static void linkAndIncludeFile(File file, String entryName,
+      ArchiveOutputStream<TarArchiveEntry> archiveOutput, Path tmpDir) throws IOException {
+    File link = null;
+    try {
+      Files.createLink(tmpDir.resolve(entryName), file.toPath());
+      link = tmpDir.resolve(file.getName()).toFile();
```

Comment:
it seems the link name is entryName rather than file
```suggestion
      link = tmpDir.resolve(entryName).toFile();
```

Reply:
done.